config/config.go

```diff
@@ -65,7 +65,7 @@ func (l *Limiter) LimitReached(key string) bool {
 		l.tokenBuckets[key] = ratelimit.NewBucket(l.TTL, l.Max)
 	}
 
-	_, isSoonerThanMaxWait := l.tokenBuckets[key].TakeMaxDuration(1, l.TTL)
+	_, isSoonerThanMaxWait := l.tokenBuckets[key].TakeMaxDuration(1, 0)
```

Comment:
Maybe I should rename that variable to something like"ableToGetMore". I think the API `LimitReached` here means whether at this moment, another token can be taken from the bucket. And here is the difference by changing `l.TTL` to `0`:

* Previous: `TakeMaxDuration(1, l.TTL)` asks wether it can take 1 more token with the max waiting time `l.TTL`.  It's not asking about current available token, instead, it "overdraft" the bucket by one `l.TTL`. In the case that current available token is 0, after `l.TTL`, 1 token will be refilled as `l.TTL` is the `refillInterval`, therefore `TakeMaxDuration(1, l.TTL)` always returns true as long as currently available bucket >= 0(but in fact, we want only > 0, this is why it's off by one). 
* Now:  changing `l.TTL` simply asks the bucket's current state and to see whether it can take one more token.